examples/cifar100_cnn.py

```diff
@@ -0,0 +1,131 @@
+'''
+#Train a simple deep CNN on the CIFAR100 images dataset.
+
+It gets to 73% validation accuracy in 25 epochs, and 75% after 50 epochs.
+(it's still underfitting at that point, though).
+'''
+
+from __future__ import print_function
+import keras
+from keras.datasets import cifar100
+from keras.preprocessing.image import ImageDataGenerator
+from keras.models import Sequential
+from keras.layers import Dense, Dropout, Activation, Flatten
+from keras.layers import Conv2D, MaxPooling2D
+import os
+
+batch_size = 32
+num_classes = 100
+epochs = 100
+data_augmentation = True
+num_predictions = 200
+save_dir = os.path.join(os.getcwd(), 'saved_models')
+model_name = 'cifar100vgg.h5'
+
+# The data, split between train and test sets:
+(x_train, y_train), (x_test, y_test) = cifar100.load_data()
+print('x_train shape:', x_train.shape)
+print(x_train.shape[0], 'train samples')
+print(x_test.shape[0], 'test samples')
+
+# Convert class vectors to binary class matrices.
+y_train = keras.utils.to_categorical(y_train, num_classes)
+y_test = keras.utils.to_categorical(y_test, num_classes)
+
+model = Sequential()
```

Comment:
Please define a state-of-the-art (or relatively close) convnet for CIFAR100, using the functional API, so as to make this example distinct from the CIFAR10 one.